include/zephyr/logging/log_msg.h

```diff
@@ -560,7 +560,7 @@ do { \
 
 /* Create local variable from input variable (expect for the first (fmt) argument). */
 #define Z_LOG_LOCAL_ARG_CREATE(idx, arg) \
-	COND_CODE_0(idx, (), (Z_AUTO_TYPE Z_LOG_LOCAL_ARG_NAME(idx, arg) = (arg) + 0))
+	COND_CODE_0(idx, (), (Z_AUTO_TYPE Z_LOG_LOCAL_ARG_NAME(idx, arg) = 0?(arg):(arg)))
```

Comment:
It probably needs to stay like it is. Usage of that macro is surrounded with `TOOLCHAIN_DISABLE_GCC_WARNING(TOOLCHAIN_WARNING_POINTER_ARITH)` I'm not sure why it is not applied for clang++.

Reply:
I'm actually surprised that CI passed so maybe it can be the solution.